src/main/java/duke/core/Core.java

```diff
@@ -0,0 +1,102 @@
+package duke.core;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Core is the central class that processes and executes all commands.
+ */
+public class Core {
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isOnline;
+
+    /**
+     * Initialises a Duke application
+     */
+    public Core() {
+        tasks = new TaskList();
+        ui = new Ui();
+        isOnline = true;
+    }
+
+    private void handleCommand(String cmd) throws DukeException {
+        if (cmd.matches("^bye$")) {
```

Comment:
Perhaps switch case can be used here? Switch is usually faster and easier to read. It might be possible with the java.util.regex package and a hash table, though it might be more complicated.